File: src/main/java/frc/robot/commands/climber/ClimberMove.java
package frc.robot.commands.climber;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.subsystems.Climber;

public class ClimberMove extends CommandBase {

  private double mSpeed;

  public ClimberMove(double speed) {
    mSpeed = speed;
  }

  @Override
  public void initialize() {
    Climber.getInstance().RetractPiston();
  }

  @Override
  public void execute() {
    Climber.getInstance().setSpeed(mSpeed);
  }

  @Override
  public boolean isFinished() {
    return false;
  }

  @Override
  public void end(boolean interrupted) {
    Climber.getInstance().setSpeed(0.0);
    Climber.getInstance().ExtendPiston();
  }

}

File: src/main/kotlin/frc/robot/commands/climber/ClimberMove.kt
package frc.robot.commands.climber

import edu.wpi.first.wpilibj2.command.CommandBase
import frc.robot.subsystems.Climber

class ClimberMove(val speed: Double): CommandBase() {

  init {
    addRequirements(Climber)
  }

  override fun initialize() {
    Climber.disengageBrake()
  }

  override fun execute() {
    Climber.setMotor(speed)
  }

  override fun end(interrupted: Boolean) {
    Climber.setMotor(0.0)
    Climber.engageBrake()
  }

}